File: src/main/java/indi/dkx/laocai/bot/model/segment/TextSegment.java
package indi.dkx.laocai.bot.model.segment;

import lombok.*;
import indi.dkx.laocai.bot.model.segment.data.TextSegmentData;

/**
 * text 文本消息段
 */
@ToString(callSuper = true)
@EqualsAndHashCode(callSuper = true)
@Data
public class TextSegment extends Segment<TextSegmentData> {
    public TextSegment(String text){
        super("text", new TextSegmentData(text));
    }
}

File: laocai-bot-core/src/main/kotlin/indi/kyson/laocai/bot/core/segment/TextSegment.kt
package indi.kyson.laocai.bot.core.segment

import com.fasterxml.jackson.annotation.JsonIgnore

/**
 * 纯文本消息段。
 */
class TextSegment private constructor(override val data: Data) : Segment {

    data class Data(val text: String) {
        fun toSegment(): TextSegment = TextSegment(this)
    }

    override val type: String
        get() = "text"

    @get:JsonIgnore
    val text: String
        get() = data.text

    companion object {
        @JvmStatic
        fun of(text: String): TextSegment = TextSegment(Data(text))
    }
}
